openedx/license-manager
# feat: add property to SubscriptionPlan serializer to denote whether plan is in renewal processing window

## Description

* Adds a model property to denote whether a given `SubscriptionPlan` is in the renewal processing window (i.e., defaults to 12 hours prior to the renewal's effective datetime).
* Migrates `DateField` to `DateTimeField` to give more granularity in start/end times of plans and renewals.
* Updates all the places where we were using regular ol' dates vs. datetimes.

Link to the associated ticket: https://openedx.atlassian.net/browse/ENT-4845

## Testing considerations
- Create a renewal for a SubscriptionPlan with an effective date within 12 hours of today's date & time.
- Hit the subscription detail API endpoint for that plan and see the `is_locked_for_renewal_processing` coming back as true.
- Change the renewal's effective date to beyond 12 hours, hit the endpoint again, and see the `is_locked_for_renewal_processing`  coming back as false.

## Post-review

Squash commits into discrete sets of changes
